cmd/ftl/cmd_deploy.go

```diff
@@ -100,10 +103,54 @@ func (d *deployCmd) Run(ctx context.Context, client ftlv1connect.ControllerServi
 		return err
 	}
 
+	if d.Wait {
+		logger.Infof("Waiting for deployment %s to become ready", resp.Msg.DeploymentName)
+
+		var wg *errgroup.Group
+		wg, ctx = errgroup.WithContext(ctx)
+		wg.Go(d.checkReadiness(ctx, client, resp.Msg.DeploymentName))
+		if err := wg.Wait(); err != nil {
+			return fmt.Errorf("deployment %s failed to become ready: %w", resp.Msg.DeploymentName, err)
+		}
+	}
+
 	logger.Infof("Successfully created deployment %s", resp.Msg.DeploymentName)
 	return nil
 }
 
+func (d *deployCmd) checkReadiness(ctx context.Context, client ftlv1connect.ControllerServiceClient, deploymentName string) func() error {
+	return func() error {
+		ticker := time.NewTicker(1 * time.Second)
+		defer ticker.Stop()
```

Comment:
I'm not sure if it has the same effect you're looking for, but you might be able to use `time.After` in the select below vs. creating a `NewTicker`.

https://github.com/TBD54566975/ftl/blob/main/cmd/ftl/cmd_dev.go#L175

Reply:
i think we actually want ticker here instead of timer? both would probably work but from googling it seems like timer is more for performing a single event after a delay whereas ticker is intended for recurring events executed at an interval

🤷‍♀️